packages/framework/src/Form/Admin/Product/Parameter/ParameterFormType.php

```diff
@@ -78,8 +86,39 @@ public function configureOptions(OptionsResolver $resolver): void
             ->setDefaults([
                 'data_class' => ParameterData::class,
                 'attr' => ['novalidate' => 'novalidate'],
+                'constraints' => [
+                    new Constraints\Callback([$this, 'validateUniqueParameterName']),
+                ],
             ])
             ->setRequired(['parameter'])
             ->setAllowedTypes('parameter', [Parameter::class, 'null']);
     }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Parameter\ParameterData $parameterData
+     * @param \Symfony\Component\Validator\Context\ExecutionContextInterface $context
+     */
+    public function validateUniqueParameterName(ParameterData $parameterData, ExecutionContextInterface $context): void
+    {
+        $form = $context->getRoot();
+        $formOptions = $form->getConfig()->getOptions();
+
+        /** @var \Shopsys\FrameworkBundle\Model\Product\Parameter\Parameter|null $currentParameter */
+        $currentParameter = $formOptions['parameter'];
+
+        foreach ($parameterData->name as $locale => $name) {
+            if ($name === null) {
+                continue;
+            }
+
+            $parameter = $this->parameterFacade->findParameterByName($name, $locale);
```

Comment:
_nitpick_ maybe we could load all parameter names at once before to avoid multiple same queries?

Reply:
after the refactoring in f9a776139d it would be too complicated